airflow/models/log.py

```diff
@@ -30,8 +30,9 @@ class Log(Base):
 
     id = Column(Integer, primary_key=True)
     dttm = Column(UtcDateTime)
-    dag_id = Column(String(ID_LEN, **COLLATION_ARGS))
-    task_id = Column(String(ID_LEN, **COLLATION_ARGS))
+    dag_id = Column(StringID())
+    task_id = Column(StringID())
+    map_index = Column(Integer, server_default='-1')
```

Comment:
If we do not want the map_index value to be set when dag_id, task_id, etc are null, I think we shouldn't use `server_default`. We could use default: 
```python
map_index = Column(Integer, default=lambda:set_map_index())
```
Then workout the map_index in `set_map_index`?

Reply:
Or we could just not have a default at all! The server_default was added on some models to make migration faster, but that’s less of a goal here.